src/loaders/vue.ts

```diff
@@ -113,6 +117,21 @@ const styleLoader = vueBlockLoader({
 const scriptLoader = vueBlockLoader({
   outputLang: "js",
   type: "script",
-  exclude: [/\bsetup\b/],
+  // If the block contains some type-only Vue macros, skip the entire block
+  // e.g. skip `defineProps<...>()`, but allow `defineProps(...)`
+  exclude({ lang, attributes, blockContents }) {
+    if (lang !== "ts" || !/\bsetup\b/.test(attributes)) {
+      return false;
+    }
+
+    const CODE_COMMENT_RE = /\/\/.*\n|\/\*[\S\s]*?\*\//g;
```

Comment:
Perhaps we could use `strip-literal`?

Reply:
I am not sure if that would increase the accuracy. But, it will be slower as it will have to parse the whole document first.

So, if performance is not really a concern and/or accuracy is more important, then I think using `js-tokens` would be better. `strip-literal` uses `js-tokens` to filter the comment and literal related stuff. We would use it to check for identifiers(the macro names in this case). Since it's iterator-based, we won't have to parse the whole document when there is a macro. So, it will be better in terms of performance compared to `strip-literal`. It would also simplify the code as we would no longer need regexes.

Please let me know if we should keep it like this or go with the `js-tokens` approach.

Reply:
sure - let's use `js-tokens` then!

Reply:
Done 👍 Just to give a rough statistic, executing the `exclude` method went from ~0.1ms to ~1ms on my machine(Apple M1).